Masterbar: enable `My sites` link on theme section

sejas opened the pull request:
#### Changes proposed in this Pull Request

`My sites` button is clickable when the user is in a different Group Section, like `Reader`.
- Now, the `My sites` is clickable in any case for the desktop version. And it is a link to `My Home`.
- For the tablet/mobile version, we keep the same current behavior that toggles the visibility of the menu.

#### Screencast


https://user-images.githubusercontent.com/779993/169977546-9bf3230d-5903-4b2b-ac67-dec4855dacb3.mp4





#### Testing instructions

**Test theme section**
- On any site.
- Go to **Appearance** > **Themes** , and select any theme.
- Observe the menu on the left is hidden.
- Click on `My Sites`.
- Observe you have navigated to `My home` in your current selected site.
- This behavior is similar to going to `Reader` or your Profile and then clicking to `My Sites`.

**Test any page**
- Click on any other page of the menu, for example, Posts.
- Click on `My Sites`.
- Observe you have navigated to `My home` in your current selected site.

Closes #49847

sejas committed:
Masterbar: #49847 enable My sites link on theme section

When going to Appearance > Themes , and selecting any theme, calypso navigates to the theme section.
This section has a css to hide the Sidebar/menu on the left.
`.theme-default.is-sidebar-overflow .is-section-theme .layout__secondary`
https://d.pr/v/xZpyC2

And the My Sites is an empty button that does nothing. This commit converts the My Sites into a link for this specific section.
https://d.pr/v/0SYH0t

Closes #49847

p-jackson commented:
Part of me feels like this is actually working as designed. None of the other sections in My Sites behave this way, so why should there be an exception just because the theme showcase just happens to be full width.

On the other hand, the Reader link doesn't do this. [Clicking the Reader link always navigates to the root of the Reader](https://github.com/Automattic/wp-calypso/blob/cc176e634aa227640bc4f95c1fc93c6036dd09a9/client/layout/masterbar/logged-in.jsx#L267).

I'll comment on the initial issue.

tyxla reviewed (commented):
My $0.02: there's a "back" button that is intended to serve as a way for the user to navigate back to the sidebar navigation where they can change the site, so this was already fine in my opinion. However, I don't see any downside in having this exception if it saves an extra click when the user's intent was to change the site while on a theme page.

sejas committed:
Masterbar: #49847 Add link to 'My Sites' for desktop versions, mobile shows and hides the menu

sejas commented:
As discussed on #49847 , I'm adding the link to `My Sites` button on every page (for the desktop version).
I've updated the PR description and added a screencast to explain why I'm adding the new Viewport check at 782px.

tyxla reviewed (approved):
This actually works well and feels like an improvement to what we had before. Thanks 👍  Unless @p-jackson has reservations, I'm happy to ship this 🚀

wojtekn commented:
@sejas looks good to me and works fine on my local. I think it improves the UX 👍

p-jackson reviewed (approved):
Great! Thanks for testing at various window widths, I'd forgotten all about that.
